migration/idempotent/011-maintenance.sql

```diff
@@ -131,11 +129,10 @@ BEGIN
             ORDER BY range_end DESC
             LIMIT 1;
         END IF;
-        -- end this txn so we're not holding any locks on the catalog
-
-        SELECT current_epoch, last_update_time INTO present_epoch, last_updated FROM
-            _prom_catalog.ids_epoch LIMIT 1;
+        PERFORM _prom_catalog.initialize_current_epoch(ran_at);
```

Comment:
My understanding this is necessary because the first `_prom_catalog.delete_expired_series` is conditional making it possible for `_prom_catalog.mark_series_to_be_dropped_as_unused` to start on `current_epoch` 0. 

It would be easier for me to reason about this if `_prom_catalog.mark_series_to_be_dropped_as_unused` simply punted when `current_epoch` is 0, letting either the second `_prom_catalog.delete_expired_series` or the connector initialize it.

Reply:
I'm not sure if that really works.

Part of the problem is that there is no expired series cleanup process. If we don't mark a series as unused in `_prom_catalog.mark_series_to_be_dropped_as_unused`, but then remove all associated chunks for the series in the subsequent call to `_prom_catalog.drop_metric_chunk_data`, then the empty series will never be removed.

If we punt in `_prom_catalog.mark_series_to_be_dropped_as_unused` then this will happen. Granted, the likelihood is low, but it could still happen.